Source/JavaScriptCore/runtime/IntlDurationFormat.cpp

```diff
@@ -411,7 +411,7 @@ static Vector<Element> collectElements(JSGlobalObject* globalObject, const IntlD
 
         // 3.l. If value is not 0 or display is not "auto", then
         value = purifyNaN(value);
-        if (value != 0 || unitData.display() != IntlDurationFormat::Display::Auto) {
+        if (value || unitData.display() != IntlDurationFormat::Display::Auto || unitData.style() == IntlDurationFormat::UnitStyle::TwoDigit || unitData.style() ==  IntlDurationFormat::UnitStyle::Numeric) {
```

Comment:
nit: How about this?
```suggestion
        auto style = unitData.style();
        if (value || unitData.display() != IntlDurationFormat::Display::Auto || style == IntlDurationFormat::UnitStyle::TwoDigit || style == IntlDurationFormat::UnitStyle::Numeric) {
```
This would allow us to replace other `unitData.style()`s below as well.